Week_4_Memory/filter-more/helpers.c

```diff
@@ -1,25 +1,152 @@
 #include "helpers.h"
+#include "math.h"
+#include "stdio.h"
 
 // Convert image to grayscale
 void grayscale(int height, int width, RGBTRIPLE image[height][width])
 {
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width; j++)
+        {
+            int blue = image[i][j].rgbtBlue;
+            int green = image[i][j].rgbtGreen;
+            int red = image[i][j].rgbtRed;
+            int average = round((blue + green + red) / 3.f);
+            image[i][j].rgbtBlue = average;
+            image[i][j].rgbtGreen = average;
+            image[i][j].rgbtRed = average;
+        }
+    }
+
     return;
 }
 
 // Reflect image horizontally
 void reflect(int height, int width, RGBTRIPLE image[height][width])
 {
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width / 2; j++)
+        {
+            int temp_blue = image[i][j].rgbtBlue;
+            int temp_green = image[i][j].rgbtGreen;
+            int temp_red = image[i][j].rgbtRed;
+
+            image[i][j].rgbtBlue = image[i][width - 1 - j].rgbtBlue;
+            image[i][j].rgbtGreen = image[i][width - 1 - j].rgbtGreen;
+            image[i][j].rgbtRed = image[i][width - 1 - j].rgbtRed;
+            image[i][width - 1 - j].rgbtBlue = temp_blue;
+            image[i][width - 1 - j].rgbtGreen = temp_green;
+            image[i][width - 1 - j].rgbtRed = temp_red;
+        }
+    }
+
     return;
 }
 
 // Blur image
 void blur(int height, int width, RGBTRIPLE image[height][width])
 {
+    RGBTRIPLE image_copy[height][width];
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width; j++)
+        {
+            image_copy[i][j] = image[i][j];
+        }
+    }
+
+    for (int row = 0; row < height; row++)
+    {
+        for (int col = 0; col < width; col++)
+        {
+            int sum_blue = 0;
+            int sum_green = 0;
+            int sum_red = 0;
+            float count = 0.0;
+
+            for (int i = -1; i < 2; i++)
```

Comment:
despite I've used iteration from 0 to 3 I prefer your approach more with `-1` which seems a bit more intuitive ❤️
<img width="484" alt="image" src="https://github.com/khaletska/CS50/assets/13601748/93bdea12-f5b9-4067-9013-3fc9f4a2610d">